website/docs/artist_hosts_maya_xgen.md

```diff
@@ -4,26 +4,92 @@ title: Xgen for Maya
+- Resource files (`.ptx`, `.xuv`) - this contains Xgen side car files used in the collection and descriptions.
+
+## Load
+
+Open the Loader tool, `OpenPype > Loader...`, and navigate to the published Xgen version. On right-click you'll get the option `Reference Xgen (ma)`
+When loading an Xgen version the following happens:
+
+- References in the Maya file.
+- Copies the Xgen file (`.xgen`) to the current workspace.
+- Modifies the Xgen file copy to load the current workspace first then the published Xgen collection.
+- Makes a custom attribute on the Xgen collection, `float_ignore`, which can be seen under the `Expressions` tab of the `Xgen` UI. This is done to initialize the Xgen delta file workflow.
+- Setup an Xgen delta file (`.xgd`) to store any workspace changes of the published Xgen version.
+
+When the loading is done, Xgen collection will be in the Xgen delta file workflow which means any changes done in the Maya workfile will be stored in the current workspace. The published Xgen collection will remain intact, even if the user assigns maps to any attributes or otherwise modifies any attribute.
+
+### Updating
+
+When there are changes to the Xgen version, the user will be notified when opening the workfile or publishing. Since the Xgen is referenced, it follows the standard Maya referencing system and overrides.
+
+For example publishing `xgenMain` version 1 with the attribute `renderer` set to `None`, then version 2 has `renderer` set to `Arnold Renderer`. When updating from version 1 to 2, the `renderer` attribute will be updated to `Arnold Renderer` unless there is an override.
+
+### Connect Patches
+
+When loading in an Xgen version, it does not have any connections to anything in the workfile, so its static in the position it was published in. Use the [Connect Geometry](artist_hosts_maya#connect-geometry) action to connect Xgen to any matching loaded animated geometry.
+
+### Connect Guides
+
+Along with patches you can also connect the Xgen guides to an Alembic cache.
+
+#### Usage
+
+Select 1 animation container, of family `animation` or `pointcache`, then the Xgen containers to connect to. Right-click > `Actions` > `Connect Xgen`.
+
+***Note: Only alembic (`.abc`) representations are allowed.***
+
+#### Details
+
+Connecting the guide will make Xgen use the Alembic directly, setting the attributes under `Guide Animation`, so the Alembic needs to contain the same amount of curves as guides in the Xgen.
+
+The animation container gets connected with the Xgen container, so if the animation container is updated so will the Xgen container's attribute.
+
+## Rendering
+
+To render with Xgen, follow the [Rendering With OpenPype](artist_hosts_maya#rendering-with-openpype) guide.
+
+### Details
+
+When submitting a workfile with Xgen, all Xgen related files will be collected and published as the workfiles resources. This means the published workfile is no longer referencing the workspace Xgen files.
```

Comment:
Is this always true? or only when use published file for rendering is enabled in the openpype settings?

Reply:
Hmm, this happens when there is an active renderlayer present in the context, but dont think it'll check for whether the rendering process is using the published file or not. Will investigate.

Reply:
Will account for not using the published file now.